
---
title: "R_Figures"
author: "Paloma Gea"
date: "`r Sys.Date()`"
output: html_document
---

Graph a scatter plot of the length of the pilS region and number of pilS cassettes from a csv file\
Set disease assoicated isolates as oranges Xs and carriage associated isolates as blue Os\

Use the csv file "pilS_comparisons.csv containing the length of the pilS region (inlcuding fbp and lpxC without the 1kb buffer) and number of silent cassettes  \
This file was generated manually based on the consensus annotations and has been included in the repository


```{r, out.width="50%", fig.width=10, fig.height=5, dpi=300, echo = FALSE}
library(ggplot2) 

###Set the variables###
pilS_comparison_csv <- "/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/pilS_comparison.csv #csv file path only needs to be changed if moved

pilS_comparison = read.table(pilS_comparison_csv, header=T, sep = ",") 
pilS_comparison$Length.of.PilS.Region.including.fbp.and.lpxC..bp.. <- 
  as.numeric(gsub("[^0-9.]", "", pilS_comparison$Length.of.PilS.Region.including.fbp.and.lpxC..bp..)) 
pilS_comparison$Number.of.other.truncated.copies. <- 
  as.numeric(gsub("[^0-9.]", "", pilS_comparison$Number.of.other.truncated.copies.)) 
pilS_plot <- ggplot(pilS_comparison, aes(x = Length.of.PilS.Region.including.fbp.and.lpxC..bp.., y = Number.of.other.truncated.copies., color = Type, shape = Type)) +
  geom_point(size = 5, stroke = 1.5) +
  scale_color_manual(values = c("c" = "#0077bb", "d" = "#ee7733"), labels = c("Carriage", "Disease")) +
  scale_shape_manual(values = c("c" = 1, "d" = 4), labels = c("Carriage", "Disease")) +
  labs(
    color = "Category",
    shape = "Category",
    title = "Length of pilS Region and Number of Silent pilS Copies",
    x = "Length (bp) pilS Region",
    y = "Number of Silent pilS Copies"
  ) +
  theme_bw() +
  theme(
    axis.title = element_text(size = 16),
    axis.text = element_text(size = 16),
    plot.title = element_text(size = 18),
    legend.title = element_text(size = 15),
    legend.text = element_text(size = 14)
  )
plot(pilS_plot)
```


Make scatter plot to show number of silent pilS copies per isolate, organised into families\
Set disease assoicated isolates as oranges Xs and carriage associated isolates as blue Os\

Use the csv file "pilS_comparisons.csv containing the length of the pilS region (inlcuding fbp and lpxC without the 1kb buffer) and number of silent cassettes  \
This file was generated manually based on the consensus annotations and has been included in the repository


```{r, out.width="50%", fig.width=10, fig.height=5, dpi=300, echo = FALSE}
library(ggplot2) 

###Set the variables###
pilS_comparison_csv <- "/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/pilS_comparison.csv #csv file path only needs to be changed if moved

pilS_comparison = read.table(pilS_comparison_csv, header=T, sep = ",") 
pilS_comparison$Number.of.other.truncated.copies. <- 
  +    as.numeric(gsub("[^0-9.]", "", pilS_comparison$Number.of.other.truncated.copies.)) 
pilS_comparison$Isolate.. <- factor(pilS_comparison$Isolate.., levels = unique(pilS_comparison$Isolate)) 
pilS_plot2 <- ggplot(pilS_comparison, aes(x = Isolate.., y = Number.of.other.truncated.copies., shape = Type, color = Type)) +
  geom_point(size = 5, stroke = 1.5) +
  scale_color_manual(values = c("#0077bb", "#ee7733"), labels = c("Carriage", "Disease")) +
  scale_shape_manual(values = c(1, 4), labels = c("Carriage", "Disease")) +
  labs(color = "Category", shape = "Category", title = "Number of Silent pilS Copies per Isolate", x = "Isolate", y = "Number of Silent pilS Copies") +
  theme_bw() +
  scale_y_continuous(limits = c(0, 13)) + 
  theme(axis.title = element_text(size = 16),axis.text = element_text(size = 16), plot.title = element_text(size = 18), legend.title = element_text(size = 15), legend.text = element_text(size = 14), axis.text.x = element_text(size = 14, angle = 45, hjust = 1))

plot(pilS_plot2)
```


Make boxplot to show the depth across the whole genome for each isolate \
Y axis has a limit of 200 \
Disease associated isolates are orange and carriage associated isolates are blue \

The coverage_txt file generated in the after_correction.sh script is used for each isolate. 

```{r, out.width="50%", fig.width=13, fig.height=8, dpi=300, echo = FALSE}
library (ggplot2)
library(dplyr)

###set variables### 
#need to set the path to the coverage.txt files for each isolate 
NZ96294_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ96294/NZ96294_coverage.txt") 
NZ97019_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ97019/NZ97019_coverage.txt") 
NZ97021_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ97021/NZ97021_coverage.txt") 
NZ97052_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ97052/NZ97052_coverage.txt") 
NZ97133_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ97133/NZ97133_coverage.txt") 
NZ97192_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ97192/NZ97192_coverage.txt") 
NZ98058_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ98058/NZ98058_coverage.txt") 
NZ98074_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ98074/NZ98074_coverage.txt") 
NZCM111_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM111/NZCM111_coverage.txt") 
NZCM112_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM112/NZCM112_coverage.txt") 
NZCM132_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM132/NZCM132_coverage.txt") 
NZCM133_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM133/NZCM133_coverage.txt") 
NZCM134_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM134/NZCM134_coverage.txt") 
NZCM135_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM135/NZCM135_coverage.txt") 
NZCM148_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM148/NZCM148_coverage.txt") 
NZCM149_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM149/NZCM149_coverage.txt") 
NZCM162_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM162/NZCM162_coverage.txt") 
NZCM165_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM165/NZCM165_coverage.txt") 
NZCM238_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM238/NZCM238_coverage.txt") 
NZCM239_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM239/NZCM239_coverage.txt") 
NZCM240_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM240/NZCM240_coverage.txt")
NZCM245_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM245/NZCM245_coverage.txt")
NZCM246_depth = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM246/NZCM246_coverage.txt")
col_names = c("contig", "position", "depth") 
colnames(NZ96294_depth) <- col_names 
colnames(NZ97019_depth) <- col_names 
colnames(NZ97021_depth) <- col_names 
colnames(NZ97052_depth) <- col_names 
colnames(NZ97133_depth) <- col_names 
colnames(NZ97192_depth) <- col_names 
colnames(NZ98058_depth) <- col_names 
colnames(NZ98074_depth) <- col_names 
colnames(NZCM111_depth) <- col_names 
colnames(NZCM112_depth) <- col_names 
colnames(NZCM132_depth) <- col_names 
colnames(NZCM133_depth) <- col_names 
colnames(NZCM134_depth) <- col_names 
colnames(NZCM135_depth) <- col_names 
colnames(NZCM148_depth) <- col_names 
colnames(NZCM149_depth) <- col_names 
colnames(NZCM162_depth) <- col_names 
colnames(NZCM165_depth) <- col_names 
colnames(NZCM238_depth) <- col_names 
colnames(NZCM239_depth) <- col_names 
colnames(NZCM240_depth) <- col_names
colnames(NZCM245_depth) <- col_names 
colnames(NZCM246_depth) <- col_names
all_isolates_list = list(NZ96294 = NZ96294_depth, NZ97019 = NZ97019_depth, NZ97021 = NZ97021_depth, NZ97052 = NZ97052_depth, NZ97133 = NZ97133_depth, NZ97192 = NZ97192_depth, NZ98058 = NZ98058_depth, NZ98074 = NZ98074_depth, NZCM111 = NZCM111_depth, NZCM112 = NZCM112_depth, NZCM132 = NZCM132_depth, NZCM133 = NZCM133_depth, NZCM134 = NZCM134_depth, NZCM135 = NZCM135_depth, NZCM148 = NZCM148_depth, NZCM149 = NZCM149_depth, NZCM162 = NZCM162_depth, NZCM165 = NZCM165_depth, NZCM238 = NZCM238_depth, NZCM239 = NZCM239_depth, NZCM240 = NZCM240_depth, NZCM245 = NZCM245_depth, NZCM246 = NZCM246_depth)
combined_data <- lapply(seq_along(all_isolates_list), function(i) {
  df <- all_isolates_list[[i]]
  df$isolate <- names(all_isolates_list)[i]
  df
})
combined_data <- do.call(rbind, combined_data)
type_values <- ifelse(grepl("CM", combined_data$isolate), "c", "d")
combined_data_coloured <- cbind(combined_data, type = type_values)
all_isolates_WG_DOC_plot <- ggplot(combined_data_coloured, aes(x = isolate, y = depth, fill = type)) + 
  geom_boxplot(position = position_dodge(width = 0.8), alpha = 0.6, outlier.shape = ".") + 
  labs(x = "Isolate", y = "Depth", title = "Depth at Each Genomic Position for Each Isolate") + 
  theme_bw() +
  theme(axis.text.x = element_text(size = 18, angle = 45, hjust = 1), axis.title = element_text(size = 20), axis.text = element_text(size = 18), plot.title = element_text(size = 22), legend.title = element_text(size = 15), legend.text = element_text(size = 14)) +
  scale_fill_manual(values = c("c" = "#0077bb", "d" = "#ee7733"), labels = c("Carriage", "Disease")) +
  scale_y_continuous(limits = c(0, 200), breaks = seq(0, 200, by = 10)) +
  guides(fill = guide_legend(title = "Isolate Type", override.aes = list(fill = c("#0077bb", "#ee7733"), labels = c("Carriage", "Disease")))) +
  stat_summary(fun=mean, geom="point", shape=18, size=3, color="black", fill="black")
plot(all_isolates_WG_DOC_plot)
```

Make Bar Graph to show number of reads mapping pilS for each isolate \
Disease associated isolates are blue and carriage associated isolates are orange\

Use the csv file "assembly_stats.csv" contaning the average depth across the genome and the reads mapping pilS for each isolate \
This file was generated manually from the output of the after_correction.sh script and has been included in the repository


```{r, out.width="50%", fig.width=10, fig.height=5, dpi=300, echo = FALSE}
library(ggplot2)
library(dplyr)

###Set the variables###
assembly_stats_csv <- "home/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/ #csv file path only needs to be changed if moved

assembly_stats = read.table(assembly_stats_csv, header=T, sep = ",") 
assembly_stats_fixed = as.data.frame(apply(assembly_stats, 2, function(x) gsub("\\st", "", x)))
assembly_stats_fixed = assembly_stats_fixed %>%
  mutate(type = ifelse(grepl("CM", Isolate), 'c', 'd')) 
assembly_stats_fixed$Reads_Mapping_PilS <- as.numeric(as.character(assembly_stats_fixed$Reads_Mapping_PilS))

reads_mapping_pilS = ggplot(assembly_stats_fixed, aes(x = Isolate, y = Reads_Mapping_PilS, fill = type)) +
  geom_col() +
  theme_bw() +
  scale_fill_manual(values = c('c' = "#0077bb", 'd' = "#ee7733"), labels = c('c' = 'Carriage', 'd' = 'Disease')) +
  scale_y_continuous(limits = c(0, 150), breaks = seq(0, 150, by = 10)) +
  labs(x = "Isolate", y = "Number of Reads", title = "Reads Mapping pilS", fill = "Type") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
        axis.text.y = element_text(size = 12),
        axis.title.x = element_text(size = 14),
        axis.title.y = element_text(size = 14),
        plot.title = element_text(size = 16),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14)) 

print(reads_mapping_pilS)


```


Make boxplot of the average read lengths mapped for each isolate \
Disease associated isolates are in orange and carriage associated isolates are in blue \ 

Use the listreadlengths.txt file generated for each isolate in the after_corrections.sh script 

```{r, out.width="50%", fig.width=30, fig.height=12, dpi=300, echo = FALSE}
library (ggplot2)
library(dplyr)
library(patchwork)

###set variables### 
#need to set the path to the coverage.txt files for each isolate 

NZ96294_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ96294/listreadlengths.txt") 
NZ97019_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ97019/listreadlengths.txt") 
NZ97021_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ97021/listreadlengths.txt") 
NZ97052_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ97052/listreadlengths.txt") 
NZ97133_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ97133/listreadlengths.txt") 
NZ97192_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ97192/listreadlengths.txt") 
NZ98058_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ98058/listreadlengths.txt") 
NZ98074_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ98074/listreadlengths.txt") 
NZCM111_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM111/listreadlengths.txt") 
NZCM112_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM112/listreadlengths.txt") 
NZCM132_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM132/listreadlengths.txt") 
NZCM133_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM133/listreadlengths.txt") 
NZCM134_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM134/listreadlengths.txt") 
NZCM135_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM135/listreadlengths.txt") 
NZCM148_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM148/listreadlengths.txt") 
NZCM149_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM149/listreadlengths.txt") 
NZCM162_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM162/listreadlengths.txt") 
NZCM165_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM165/listreadlengths.txt") 
NZCM238_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM238/listreadlengths.txt") 
NZCM239_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM239/listreadlengths.txt") 
NZCM240_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM240/listreadlengths.txt")
NZCM245_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM245/listreadlengths.txt")
NZCM246_read_length = read.table("/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZCM246/listreadlengths.txt")
col_names_read_length = c("read_length")
colnames(NZ96294_read_length) <- col_names_read_length
colnames(NZ97019_read_length) <- col_names_read_length
colnames(NZ97021_read_length) <- col_names_read_length
colnames(NZ97052_read_length) <- col_names_read_length
colnames(NZ97133_read_length) <- col_names_read_length
colnames(NZ97192_read_length) <- col_names_read_length
colnames(NZ98058_read_length) <- col_names_read_length
colnames(NZ98074_read_length) <- col_names_read_length
colnames(NZCM111_read_length) <- col_names_read_length
colnames(NZCM112_read_length) <- col_names_read_length
colnames(NZCM132_read_length) <- col_names_read_length
colnames(NZCM133_read_length) <- col_names_read_length
colnames(NZCM134_read_length) <- col_names_read_length
colnames(NZCM135_read_length) <- col_names_read_length
colnames(NZCM148_read_length) <- col_names_read_length
colnames(NZCM149_read_length) <- col_names_read_length
colnames(NZCM162_read_length) <- col_names_read_length
colnames(NZCM165_read_length) <- col_names_read_length
colnames(NZCM238_read_length) <- col_names_read_length
colnames(NZCM239_read_length) <- col_names_read_length
colnames(NZCM240_read_length) <- col_names_read_length
colnames(NZCM245_read_length) <- col_names_read_length
colnames(NZCM246_read_length) <- col_names_read_length
all_isolates_list_read_length = list(NZ96294 = NZ96294_read_length, NZ97019 = NZ97019_read_length, NZ97021 = NZ97021_read_length, NZ97052 = NZ97052_read_length, NZ97133 = NZ97133_read_length, NZ97192 = NZ97192_read_length, NZ98058 = NZ98058_read_length, NZ98074 = NZ98074_read_length, NZCM111 = NZCM111_read_length, NZCM112 = NZCM112_read_length, NZCM132 = NZCM132_read_length, NZCM133 = NZCM133_read_length, NZCM134 = NZCM134_read_length, NZCM135 = NZCM135_read_length, NZCM148 = NZCM148_read_length, NZCM149 = NZCM149_read_length, NZCM162 = NZCM162_read_length, NZCM165 = NZCM165_read_length, NZCM238 = NZCM238_read_length, NZCM239 = NZCM239_read_length, NZCM240 = NZCM240_read_length, NZCM245 = NZCM245_read_length, NZCM246 = NZCM246_read_length)
combined_data_read_length <- lapply(seq_along(all_isolates_list_read_length), function(i) {
  df <- all_isolates_list_read_length[[i]]
  df$isolate <- names(all_isolates_list_read_length)[i]
  df
})
combined_data_read_length <- do.call(rbind, combined_data_read_length)
type_values_read_length <- ifelse(grepl("CM", combined_data_read_length$isolate), "c", "d")
combined_data_read_length_coloured <- cbind(combined_data_read_length, type = type_values_read_length)
max_read_lengths <- combined_data_read_length_coloured %>%
  group_by(isolate) %>%
  summarise(Max = max(read_length))
combined_data_read_length_coloured_max_values <- combined_data_read_length_coloured %>%
  left_join(max_read_lengths, by='isolate')
get_box_stats <- function(y, upper_limit = max(combined_data_read_length_coloured$read_length) * 1.15) {
  return(data.frame(
    y = 0.95 * upper_limit,
    label = paste(
      "Count =", length(y), "\n",
      "Mean =", round(mean(y), 2), "\n",
      "Median =", round(median(y), 2), "\n"
    )
  ))
}
all_isolates_read_length_plot <- ggplot(combined_data_read_length_coloured_max_values, aes(x = isolate, y = read_length, fill = type)) +
  geom_boxplot(position = position_dodge(width = 0.8), alpha = 0.6, outlier.shape = ".") +
  labs(x = "Isolate", y = "Read Length", title = "Read Lengths for Each Isolate") +
  theme_bw() +
  theme(axis.text.x = element_text(size = 18, angle = 45, hjust = 1), axis.title = element_text(size = 25), axis.text = element_text(size = 18), plot.title = element_text(size = 30), legend.title = element_text(size = 25), legend.text = element_text(size = 20)) +
  scale_fill_manual(values = c("c" = "#0077bb", "d" = "#ee7733"), labels = c("Carriage", "Disease")) +
  scale_y_continuous() +
  guides(fill = guide_legend(title = "Isolate Type", override.aes = list(fill = c("#0077bb", "#ee7733"), labels = c("Carriage", "Disease")))) +
  geom_text(data = combined_data_read_length_coloured_max_values, aes(x = isolate, y = Max, label = sprintf("%.0f", Max)), vjust = -0.5, size = 5, color = "#565052")
all_isolates_read_length_plot_zoomed = ggplot(combined_data_read_length_coloured_max_values, aes(x = isolate, y = read_length, fill = type)) +
  geom_boxplot(position = position_dodge(width = 0.8), alpha = 0.6, outlier.shape = ".") +
  labs(x = "Isolate", y = "Read Length (bp)", title = "Read Lengths for Each Isolate") +
  theme_bw() +
  theme(axis.text.x = element_text(size = 25, angle = 45, hjust = 1), axis.title = element_text(size = 30), axis.text = element_text(size = 25), plot.title = element_text(size = 35), legend.title = element_text(size = 25), legend.text = element_text(size = 25)) +
  scale_fill_manual(values = c("c" = "#0077bb", "d" = "#ee7733"), labels = c("Carriage", "Disease")) +
  scale_y_continuous(limits = c(0, 20000), breaks = seq(0, 19000, by = 1000)) +
  guides(fill = guide_legend(title = "Isolate Type", override.aes = list(fill = c("#0077bb", "#ee7733"), size = 18, labels = c("Carriage", "Disease")))) +
 geom_text(data = combined_data_read_length_coloured_max_values, aes(x = isolate, y = 19500, label = sprintf("%.0f", Max)), vjust = -1, size = 8, color = "#565052") +
  stat_summary(fun=mean, geom="point", shape=18, size=3, color="black", fill="black")

plot(all_isolates_read_length_plot)
plot(all_isolates_read_length_plot_zoomed)
```


Graph to show the change in depth across the pilS region to identify isolates with a deletion in this region\
All reads spanning across pilS were extracted and the depth calculated at each position in pilS\
Depth should be constant throughout the pilS region; change in depth indicateds a deletion

Isolate and path to the pilS_depths.txt file need to be set 

```{r, out.width="90%", fig.width=20, fig.height=13, dpi=300, echo = FALSE}
library(ggplot2) 
library(patchwork)

###Set the variables###
isolate <- "NZ96294" #Change for each isolate 
pilS_depth_txt <- "/Genomic-Organisation-of-MeningococcaL-pilS-in-Carriage-and-Disease/NZ96294/NZ96294_pilS_depths.txt" #Change to the path where the pilS_depths.txt file is stored for that isolate

pilS_depth = read.table(pilS_depth_txt, sep="\t", header=FALSE, col.names=c("contig", "position", "depth")) 
pilS_depth$position_adjusted = pilS_depth$position - min(pilS_depth$position) 
pilS_depth_plot = ggplot(pilS_depth, aes(x=position_adjusted, y=depth, group=contig, color=contig)) + geom_line(color = "#ee7733") + labs(title="Depth of Coverage Across pilS", x="Position", y="Depth") + theme_minimal() + theme(axis.text = element_text(size = 10), legend.position = "none")  

plot(pilS_depth_plot)

```


Deletions across genome \
Plot the variance at each window across the genome \
Add red bars showing the mean variance across the whole genome and two standard devaitions above and below this \
Variance outside this range was considered statistically significant \
Each window is plotted based on the start position of the window ie window 0_10000 is plotted at genomic positon 0

Graphs can be made for each isolate and use the averages_per_window.txt file generated in the deletions_across_genome.sh script

```{r, out.width="50%", fig.width=12, fig.height=8, dpi=300, echo = FALSE}
library(ggplot2)
library(patchwork)

#Load the averages_per_window.txt file for each isolate and save as a variable name, ensure correct path is set
NZ96294_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZ96294/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZCM148_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZCM148/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZCM149_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZCM149/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZ97019_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZ97019/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZCM162_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZCM162/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZ97021_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZ97021/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZCM165_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZCM165/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZ97052_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZ97052/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZCM111_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZCM111/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZCM112_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZCM112/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZ97133_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZ97133/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZCM132_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZCM132/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZCM133_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZCM133/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZCM134_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZCM134/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZCM135_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZCM135/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZ97192_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZ97192/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZCM238_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZCM238/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZCM239_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZCM239/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZCM240_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZCM240/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZ98058_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZ98058/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZCM246_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZCM246/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZ98074_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZ98074/deletions/averages_per_window.txt", sep="\t", header=TRUE)
NZCM245_deletions_windows = read.table("/Genomic-Organisation-of-Meningococcal-pilS-in-Carriage-and-Disease/NZCM245/deletions/averages_per_window.txt", sep="\t", header=TRUE)

###Set variables###
deletions_windows = NZ96294_deletions_windows #set which isolate you are building graph for this example is isolate NZ96294 
pilS_start <- 625149 #set the pilS region for the specific isolate this example is NZ96294
pilS_end <- 633833 #set the pilS region for the specific isolate this example is NZ96294

deletions_windows = read.table(averages_per_window, sep="\t", header=TRUE)
start_positions <- gsub("window_(\\d+)_(\\d+)", "\\1", deletions_windows$window)
start_positions <- as.numeric(start_positions)
deletions_windows$position <- start_positions
mean_value_of_variance <- mean(deletions_windows$sample_variance)
sd_value_of_variance <- sd(deletions_windows$sample_variance)
deletions_var_plot = ggplot(deletions_windows, aes(x = position, y = sample_variance)) +
  geom_line(size = 0.3) +
  geom_hline(yintercept = mean_value_of_variance, linetype = "dotted", color = "#ee7733") +
  geom_ribbon(aes(ymin = mean_value_of_variance - 2 * sd_value_of_variance, ymax = mean_value_of_variance + 2 *  sd_value_of_variance), fill = "#ee7733AA", alpha = 0.2) +
  scale_x_continuous(labels = scales::comma_format(scale = 1, accuracy = 1), expand = c(0, 0)) +
  scale_y_continuous(expand = c(0, 0)) + coord_cartesian(ylim = c(0, 300)) +
  labs(x = "Genomic Position", y = "Variance", title = "Variance of Depth of Coverage per 10kb Windows") +
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 14),
    axis.text.y = element_text(size = 14),
    axis.title.x = element_text(size = 16),
    axis.title.y = element_text(size = 16),
    plot.title = element_text(size = 18),
    axis.line = element_line())+
  coord_cartesian(ylim = c(0,100))
## Add a blue bar to show the position of the pilS region 
deletions_var_plot_2 = ggplot(deletions_windows, aes(x = position, y = sample_variance)) +
  geom_line(size = 0.3) +
  geom_hline(yintercept = mean_value_of_variance, linetype = "dotted", color = "#ee7733") +
  geom_ribbon(aes(ymin = mean_value_of_variance - 2 * sd_value_of_variance, ymax = mean_value_of_variance + 2 *  sd_value_of_variance), fill = "#ee7733AA", alpha = 0.2) +
  geom_rect(aes(xmin = pilS_start, xmax = pilS_end, ymin = -Inf, ymax = Inf), fill = "#0077bbAA") +
  scale_x_continuous(labels = scales::comma_format(scale = 1, accuracy = 1), expand = c(0, 0)) +
  scale_y_continuous(expand = c(0, 0)) + coord_cartesian(ylim = c(0, 300)) +
  labs(x = "Genomic Position", y = "Variance", title = "Variance of Depth of Coverage per 10kb Windows") +
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 14),
    axis.text.y = element_text(size = 14),
    axis.title.x = element_text(size = 16),
    axis.title.y = element_text(size = 16),
    plot.title = element_text(size = 18),
    axis.line = element_line())+
  coord_cartesian(ylim = c(0,100))
##zoom in on NZ96294 pilS
deletions_var_plot_zoomed = ggplot(deletions_windows, aes(x = position, y = sample_variance)) +
  geom_line(size = 0.3) +
  geom_hline(yintercept = mean_value_of_variance, linetype = "dotted", color = "#ee7733") +
  geom_ribbon(aes(ymin = mean_value_of_variance - 2 * sd_value_of_variance, ymax = mean_value_of_variance + 2 *  sd_value_of_variance), fill = "#ee7733AA", alpha = 0.2) +
  scale_x_continuous(labels = scales::comma_format(scale = 1, accuracy = 1), expand = c(0, 0)) +
  scale_y_continuous(expand = c(0, 0)) +
  labs(x = "Genomic Position", y = "Variance", title = "Variance of Depth of Coverage per 10kb Windows zoomed on pilS +/- 1kb") +
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 14),
    axis.text.y = element_text(size = 14),
    axis.title.x = element_text(size = 16),
    axis.title.y = element_text(size = 16),
    plot.title = element_text(size = 18),
    axis.line = element_line()
  ) +
  coord_cartesian(xlim = c(pilS_start - 10000, pilS_end + 10000))  # zoom on the pilS region
(deletions_var_plot / deletions_var_plot_2 / deletions_var_plot_zoomed)
```


Make a box plot of the variance of each 10kb window for each isolate \
The windows containing the pilS region are coloured, the other windows are grey \
Makes two plots, one organised by carriage vs disease and the other organised in household groups \

This script requires that a deletions_windows file has been made for each isolate (see script above) and is saved in the same environment

```{r, out.width="50%", fig.width=18, fig.height=13, dpi=300, echo = FALSE}
library (ggplot2)
library(dplyr)
library(patchwork)


col_names_deletions_windows = c("window","mean","mode","median","sample_variance","standard_deviation","minimum","maximum","position")
colnames(NZ96294_deletions_windows) <- col_names_deletions_windows
colnames(NZ97019_deletions_windows) <- col_names_deletions_windows
colnames(NZ97021_deletions_windows) <- col_names_deletions_windows
colnames(NZ97052_deletions_windows) <- col_names_deletions_windows
colnames(NZ97133_deletions_windows) <- col_names_deletions_windows
colnames(NZ97192_deletions_windows) <- col_names_deletions_windows
colnames(NZ98058_deletions_windows) <- col_names_deletions_windows
colnames(NZ98074_deletions_windows) <- col_names_deletions_windows
colnames(NZCM111_deletions_windows) <- col_names_deletions_windows
colnames(NZCM112_deletions_windows) <- col_names_deletions_windows
colnames(NZCM132_deletions_windows) <- col_names_deletions_windows
colnames(NZCM133_deletions_windows) <- col_names_deletions_windows
colnames(NZCM134_deletions_windows) <- col_names_deletions_windows
colnames(NZCM135_deletions_windows) <- col_names_deletions_windows
colnames(NZCM148_deletions_windows) <- col_names_deletions_windows
colnames(NZCM149_deletions_windows) <- col_names_deletions_windows
colnames(NZCM162_deletions_windows) <- col_names_deletions_windows
colnames(NZCM165_deletions_windows) <- col_names_deletions_windows
colnames(NZCM238_deletions_windows) <- col_names_deletions_windows
colnames(NZCM239_deletions_windows) <- col_names_deletions_windows
colnames(NZCM240_deletions_windows) <- col_names_deletions_windows
colnames(NZCM245_deletions_windows) <- col_names_deletions_windows
colnames(NZCM246_deletions_windows) <- col_names_deletions_windows
all_isolates_list_deletions_windows = list(NZ96294 = NZ96294_deletions_windows, NZ97019 = NZ97019_deletions_windows, NZ97021 = NZ97021_deletions_windows, NZ97052 = NZ97052_deletions_windows, NZ97133 = NZ97133_deletions_windows, NZ97192 = NZ97192_deletions_windows, NZ98058 = NZ98058_deletions_windows, NZ98074 = NZ98074_deletions_windows, NZCM111 = NZCM111_deletions_windows, NZCM112 = NZCM112_deletions_windows, NZCM132 = NZCM132_deletions_windows, NZCM133 = NZCM133_deletions_windows, NZCM134 = NZCM134_deletions_windows, NZCM135 = NZCM135_deletions_windows, NZCM148 = NZCM148_deletions_windows, NZCM149 = NZCM149_deletions_windows, NZCM162 = NZCM162_deletions_windows, NZCM165 = NZCM165_deletions_windows, NZCM238 = NZCM238_deletions_windows, NZCM239 = NZCM239_deletions_windows, NZCM240 = NZCM240_deletions_windows, NZCM245 = NZCM245_deletions_windows, NZCM246 = NZCM246_deletions_windows)
combined_data_deletions_windows <- lapply(seq_along(all_isolates_list_deletions_windows), function(i) {
  df <- all_isolates_list_deletions_windows[[i]]
  df$isolate <- names(all_isolates_list_deletions_windows)[i]
  df})
combined_data_deletions_windows <- do.call(rbind, combined_data_deletions_windows)
type_values_deletions_windows <- ifelse(grepl("CM", combined_data_deletions_windows$isolate), "c", "d")
combined_data_deletions_windows_coloured <- cbind(combined_data_deletions_windows, type = type_values_deletions_windows)
combined_data_deletions_windows_coloured$pilS <- "no"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZ96294" & 
                                              combined_data_deletions_windows_coloured$position >= 619500 & 
                                              combined_data_deletions_windows_coloured$position <= 629500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZCM149" & 
                                              combined_data_deletions_windows_coloured$position >= 585500 & 
                                              combined_data_deletions_windows_coloured$position <= 595500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZCM148" & 
                                              combined_data_deletions_windows_coloured$position >= 950500 & 
                                              combined_data_deletions_windows_coloured$position <= 960500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZ97019" & 
                                              combined_data_deletions_windows_coloured$position >= 818000 & 
                                              combined_data_deletions_windows_coloured$position <= 828000] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZCM162" & 
                                              combined_data_deletions_windows_coloured$position >= 572500 & 
                                              combined_data_deletions_windows_coloured$position <= 582500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZ97021" & 
                                              combined_data_deletions_windows_coloured$position >= 514500 & 
                                              combined_data_deletions_windows_coloured$position <= 524500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZCM165" & 
                                              combined_data_deletions_windows_coloured$position >= 689500 & 
                                              combined_data_deletions_windows_coloured$position <= 699500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZ97052" & 
                                              combined_data_deletions_windows_coloured$position >= 576000 & 
                                              combined_data_deletions_windows_coloured$position <= 586000] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZCM111" & 
                                              combined_data_deletions_windows_coloured$position >= 1163000 & 
                                              combined_data_deletions_windows_coloured$position <= 1173000] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZCM112" & 
                                              combined_data_deletions_windows_coloured$position >= 1867500 & 
                                              combined_data_deletions_windows_coloured$position <= 1877500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZ97133" & 
                                              combined_data_deletions_windows_coloured$position >= 1648500 & 
                                              combined_data_deletions_windows_coloured$position <= 1658500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZCM132" & 
                                              combined_data_deletions_windows_coloured$position >= 1654500 & 
                                              combined_data_deletions_windows_coloured$position <= 1664500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZCM133" & 
                                              combined_data_deletions_windows_coloured$position >= 422500 & 
                                              combined_data_deletions_windows_coloured$position <= 432500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZCM134" & 
                                              combined_data_deletions_windows_coloured$position >= 279500 & 
                                              combined_data_deletions_windows_coloured$position <= 289500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZCM135" & 
                                              combined_data_deletions_windows_coloured$position >= 427500 & 
                                              combined_data_deletions_windows_coloured$position <= 437500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZ97192" & 
                                              combined_data_deletions_windows_coloured$position >= 285000 & 
                                              combined_data_deletions_windows_coloured$position <= 295000] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZCM238" & 
                                              combined_data_deletions_windows_coloured$position >= 1909500 & 
                                              combined_data_deletions_windows_coloured$position <= 1919500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZCM239" & 
                                              combined_data_deletions_windows_coloured$position >= 1704000 & 
                                              combined_data_deletions_windows_coloured$position <= 1714000] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZCM240" & 
                                              combined_data_deletions_windows_coloured$position >= 1619500 & 
                                              combined_data_deletions_windows_coloured$position <= 1629500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZ98058" & 
                                              combined_data_deletions_windows_coloured$position >= 1704500 & 
                                              combined_data_deletions_windows_coloured$position <= 1714500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZCM246" & 
                                              combined_data_deletions_windows_coloured$position >= 424000 & 
                                              combined_data_deletions_windows_coloured$position <= 434000] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZ98074" & 
                                              combined_data_deletions_windows_coloured$position >= 424500 & 
                                              combined_data_deletions_windows_coloured$position <= 434500] <- "yes"
combined_data_deletions_windows_coloured$pilS[combined_data_deletions_windows_coloured$isolate == "NZCM245" & 
                                              combined_data_deletions_windows_coloured$position >= 1642500 & 
                                              combined_data_deletions_windows_coloured$position <= 1652500] <- "yes"
combined_data_deletions_windows_coloured_family <- combined_data_deletions_windows_coloured
combined_data_deletions_windows_coloured_family$isolate <- factor(combined_data_deletions_windows_coloured$isolate,
                         c("NZ96294", "NZCM149", "NZCM148", "NZ97019", "NZCM162", "NZ97021", "NZCM165", "NZ97133", "NZCM132", "NZCM133", "NZCM134", "NZCM135", "NZ97192", "NZCM238", "NZCM239", "NZCM240", "NZ98058", "NZCM246", "NZ98074", "NZCM245", "NZ97052", "NZCM111", "NZCM112"))

all_isolates_deletions_windows_plot_carriage_disease <- ggplot(combined_data_deletions_windows_coloured, aes(x = isolate, y = sample_variance)) +
  geom_boxplot(outlier.shape = 1, outlier.size = 3, outlier.colour = "#C0C0C0") +
  labs(x = "Isolate", y = "Variance per Window", title = "Variance per 10bk Window of Genome per Isolate") +
  theme_bw() +
  theme(axis.text.x = element_text(size = 18, angle = 45, hjust = 1), axis.title = element_text(size = 25), axis.text = element_text(size = 18), plot.title = element_text(size = 30), legend.title = element_text(size = 25), legend.text = element_text(size = 20)) +
  scale_y_continuous(trans = 'log2')


all_isolates_deletions_windows_plot_carriage_disease + 
  geom_point(data = combined_data_deletions_windows_coloured, aes(color = pilS), 
             shape = 1, size = 3) +
  scale_color_manual(values = c("yes" = "#ee7733", "no" = "#FFFFFF00"), labels = c("Rest of Genome", "pilS")) +
  labs(color = "pilS") 

all_isolates_deletions_windows_plot_family <- ggplot(combined_data_deletions_windows_coloured_family, aes(x = isolate, y = sample_variance)) +
  geom_boxplot(outlier.shape = 1, outlier.size = 3, outlier.colour = "#C0C0C0") +
  labs(x = "Isolate", y = "Variance per Window", title = "Variance per 10bk Window of Genome per Isolate") +
  theme_bw() +
  theme(axis.text.x = element_text(size = 18, angle = 45, hjust = 1), axis.title = element_text(size = 25), axis.text = element_text(size = 18), plot.title = element_text(size = 30), legend.title = element_text(size = 25), legend.text = element_text(size = 20)) +
  scale_y_continuous(trans = 'log2')


all_isolates_deletions_windows_plot_family + 
  geom_point(data = combined_data_deletions_windows_coloured_family, aes(color = pilS), 
             shape = 1, size = 3) +
  scale_color_manual(values = c("yes" = "#ee7733", "no" = "#FFFFFF00"), labels = c("Rest of Genome", "pilS")) +
  labs(color = "pilS")

plot(all_isolates_deletions_windows_plot_carriage_disease)
plot(all_isolates_deletions_windows_plot_family)
```

Nanopore vs Illumina 
```{r, out.width="50%", fig.width=10, fig.height=8, dpi=300, echo = FALSE}

library(ggplot2)
setwd("/NGS/active/IPL/MENINGO/analysis/paloma/ONT_vs_Illumina") 

ONT_HAC_doc_by_gc = read.table("lambda_ONT_HAC_gc_by_median_cov_100_bp_window", header=T, sep = "\t") 
gc_bin_order = c('0_10', '11_20', '21_30','31_40','41_50','51_60','61_70','71_80','81_90','91_100') 
ONT_HAC_doc_by_gc$gc_bin = factor(ONT_HAC_doc_by_gc$gc_bin, levels = gc_bin_order) 
ONT_HAC_doc_by_gc_plot_final = ggplot(ONT_HAC_doc_by_gc, aes(x = gc_bin, y = median_doc)) +geom_boxplot(fill = "#ee7733", col = "black") +
  theme_bw() + labs(title="Depth of Coverage by GC Content for Lambda High Accuracy Base Called ONT", x="GC Bin", y="Median Depth of Coverage") +
  theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1), axis.title = element_text(size = 14), axis.text = element_text(size = 12), plot.title = element_text(size = 15))


ONT_SHA_doc_by_gc = read.table("lambda_ONT_SHA_gc_by_median_cov_100_bp_window", header=T, sep = "\t") 
gc_bin_order = c('0_10', '11_20', '21_30','31_40','41_50','51_60','61_70','71_80','81_90','91_100') 
ONT_SHA_doc_by_gc$gc_bin = factor(ONT_SHA_doc_by_gc$gc_bin, levels = gc_bin_order) 
ONT_SHA_doc_by_gc_plot_final = ggplot(ONT_SHA_doc_by_gc, aes(x = gc_bin, y = median_doc)) +geom_boxplot(fill = "#ee7733", col = "black") +
  theme_bw() + labs(title="Depth of Coverage by GC Content for Lambda Super High Accuracy Base Called ONT", x="GC Bin", y="Median Depth of Coverage") +
  theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1), axis.title = element_text(size = 14), axis.text = element_text(size = 12), plot.title = element_text(size = 15))


ONT_NZ97021_SHA_doc_by_gc = read.table("NZ97021_kit14_SHA_gc_by_median_cov_100_bp_window", header=T, sep = "\t") 
gc_bin_order = c('0_10', '11_20', '21_30','31_40','41_50','51_60','61_70','71_80','81_90','91_100') 
ONT_NZ97021_SHA_doc_by_gc$gc_bin = factor(ONT_NZ97021_SHA_doc_by_gc$gc_bin, levels = gc_bin_order) 
ONT_NZ97021_SHA_doc_by_gc_plot_final = ggplot(ONT_NZ97021_SHA_doc_by_gc, aes(x = gc_bin, y = median_doc)) +geom_boxplot(fill = "#ee7733", col = "black") +
  theme_bw() + labs(title="Depth of Coverage by GC Content for NZ97021 Super High Accuracy Base Called ONT", x="GC Bin", y="Median Depth of Coverage") +
  theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1), axis.title = element_text(size = 14), axis.text = element_text(size = 12), plot.title = element_text(size = 15))

ONT_NZ97019_SHA_doc_by_gc = read.table("NZ97019_kit14_SHA_gc_by_median_cov_100_bp_window", header=T, sep = "\t") 
gc_bin_order = c('0_10', '11_20', '21_30','31_40','41_50','51_60','61_70','71_80','81_90','91_100') 
ONT_NZ97019_SHA_doc_by_gc$gc_bin = factor(ONT_NZ97019_SHA_doc_by_gc$gc_bin, levels = gc_bin_order) 
ONT_NZ97019_SHA_doc_by_gc_plot_final = ggplot(ONT_NZ97019_SHA_doc_by_gc, aes(x = gc_bin, y = median_doc)) +geom_boxplot(fill = "#ee7733", col = "black") +
  theme_bw() + labs(title="Depth of Coverage by GC Content for NZ97019 Super High Accuracy Base Called ONT", x="GC Bin", y="Median Depth of Coverage") +
  theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1), axis.title = element_text(size = 14), axis.text = element_text(size = 12), plot.title = element_text(size = 15))

plot (ONT_HAC_doc_by_gc_plot_final) 
plot (ONT_SHA_doc_by_gc_plot_final)
plot(ONT_NZ97021_SHA_doc_by_gc_plot_final)
plot(ONT_NZ97019_SHA_doc_by_gc_plot_final)

HAC_GCpercent_100bp_windows_plot = ggplot(ONT_HAC_doc_by_gc, aes(x=gc_percent)) + geom_histogram(binwidth=5, color="white", fill="#ee7733") + 
  labs(title="GC Percent per 100bp Sliding Window Lambda High Accuracy Base Calling ONT", x="GC%", y="Number of 100bp Windows")  +
  theme_bw() +
  theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1), axis.title = element_text(size = 14), axis.text = element_text(size = 12), plot.title = element_text(size = 15))

SHA_GCpercent_100bp_windows_plot = ggplot(ONT_SHA_doc_by_gc, aes(x=gc_percent)) + geom_histogram(binwidth=5, color="white", fill="#ee7733") + 
  labs(title="GC Percent per 100bp Sliding Window Lambda Super High Accuracy Base Calling ONT", x="GC%", y="Number of 100bp Windows")  +
  theme_bw() +
  theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1), axis.title = element_text(size = 14), axis.text = element_text(size = 12), plot.title = element_text(size = 15))

NZ97021_SHA_GCpercent_100bp_windows_plot = ggplot(ONT_NZ97021_SHA_doc_by_gc, aes(x=gc_percent)) + geom_histogram(binwidth=5, color="white", fill="#ee7733") + 
  labs(title="GC Percent per 100bp Sliding Window NZ97021 Super High Accuracy Base Calling ONT", x="GC%", y="Number of 100bp Windows") +
  theme_bw() +
  theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1), axis.title = element_text(size = 14), axis.text = element_text(size = 12), plot.title = element_text(size = 15))

NZ97019_SHA_GCpercent_100bp_windows_plot = ggplot(ONT_NZ97019_SHA_doc_by_gc, aes(x=gc_percent)) + geom_histogram(binwidth=5, color="white", fill="#ee7733") + 
  labs(title="GC Percent per 100bp Sliding Window NZ97019 Super High Accuracy Base Calling ONT", x="GC%", y="Number of 100bp Windows") +
  theme_bw() +
  theme(axis.text.x = element_text(size = 12, angle = 45, hjust = 1), axis.title = element_text(size = 14), axis.text = element_text(size = 12), plot.title = element_text(size = 15))

plot (HAC_GCpercent_100bp_windows_plot) 
plot (SHA_GCpercent_100bp_windows_plot)
plot (NZ97021_SHA_GCpercent_100bp_windows_plot)
plot(NZ97019_SHA_GCpercent_100bp_windows_plot)


```
